addons/point_of_sale/static/src/app/hooks/use_tours.js

```diff
@@ -0,0 +1,62 @@
+import { tourState } from "@web_tour/tour_service/tour_state";
+import { TourSelectorPopup } from "../components/tour_selector_popup/tour_selector_popup";
+import { makeAwaitable } from "../store/make_awaitable_dialog";
+import { useService } from "@web/core/utils/hooks";
+
+export default function useTours() {
+    const tour = useService("tour_service");
+    const dialog = useService("dialog");
+    const states = {
+        selectedTours: new Set(),
+        running: false,
+        index: 0,
+    };
+
+    let fakeTourInterval = null;
+
+    tourState.clear();
+
+    const toggle = async () => {
+        states.index = 0;
+        states.running = !states.running;
+        clearInterval(fakeTourInterval);
+
+        if (!states.running) {
+            tourState.clear();
+            return;
+        }
+
+        const tours = await makeAwaitable(dialog, TourSelectorPopup, {});
+        if (!tours || !tours.length) {
+            tourState.clear();
+            states.running = false;
+            return;
+        }
+
+        states.selectedTours = tours;
+        fakeTourInterval = setInterval(() => {
+            const state = tourState.getCurrentTour();
+            if (!state) {
+                runTour();
+            }
+        }, 500);
```

Comment:
Why use an interval? Can you just call the startTour on the selected tour?

Reply:
I want to launch tour in loop